src/app/components/ConfirmOrCancel/index.tsx

```diff
@@ -9,7 +9,7 @@ export type Props = {
   disabled?: boolean;
   loading?: boolean;
   label?: string;
-  onConfirm: MouseEventHandler;
```

Comment:
I would still pass the confirm function to the component.
This component is used in so many places and in my eyes `onConfirm` should never be forgotten.

What do you think?
Why make it optional?

Reply:
Actually passing `onConfirm` prop to the component & using `handleSubmit()` in form at the same time leads to calling `Confirm` fxn twice which I feel should not be called twice. I logged both events when Confirm button is clicked / Enter is pressed in that form. Below mentioned are the logs for the same.

![image](https://user-images.githubusercontent.com/67408018/195400753-e6dfa252-f6c9-4755-9b9d-2da6143f7ac3.png)

Here,
first log is printed through  
![image](https://user-images.githubusercontent.com/67408018/195403198-1581ae1e-d65c-4a6e-be5b-bbe324044ceb.png)

https://github.com/getAlby/lightning-browser-extension/blob/f657162c68548f7832c16fd166083ff5e402b907/src/app/components/ConfirmOrCancel/index.tsx#L34

second log is through 

![image](https://user-images.githubusercontent.com/67408018/195403091-90aa9a2f-7106-4f01-9c07-aab00558f714.png)

![image](https://user-images.githubusercontent.com/67408018/195402869-2c202fc6-a3f9-462a-82f9-5d52d1a6da9b.png)

https://github.com/vaibhavgarg237/lightning-browser-extension/blob/7bfd5ae2dbd4a558d6a38bd688b7d461dab559ac/src/app/screens/LNURLPay/index.tsx#L349




For this, I think we need to make `onConfirm` prop optional and not pass it as prop so that it's not called twice as done in the current commit. Another workaround would be to remove `onSubmit` from `<form onSubmit={handleSubmit}>` through which we can pass the prop without making it optional but @escapedcat  suggested that it should be like [other form situations](https://github.com/getAlby/lightning-browser-extension/pull/1576#discussion_r990764381)

Reply:
Yes, this is happening because two elements are attached to the same handler.
You can avoid this with `e.stopImmediatePropagation()` 
=> https://developer.mozilla.org/en-US/docs/Web/API/Event/stopImmediatePropagation

But as discussed below, keep the `onConfirm` method optional, declare the button as type "submit"
and refactor all other occurrences of `ConfirmOrCancel` to ensure that all are used in a form element.
Otherwise this is not consistent.

Reply:
Okay, I will update the commit with these changes.

Reply:
Just one thing: As @escapedcat said below, the refactoring of the other occurrences where ConfirmOrCancel is used can be done in another extra PR. Sorry for the confusion.

Reply:
Okay. Thanks for the clarification. I will create another issue & PR for this.